crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,16 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    pub fn count_ops(&mut self) -> IndexMap<String, i32> {
+        let mut ops_count: IndexMap<String, i32> = IndexMap::new();
+        let circuit: &Vec<PackedInstruction> = &self.data;
+        for instruction in circuit {
```

Comment:
Just to make the code a bit more consise:

```suggestion
        for instruction in &self.data {
```